proxy/src/main/java/com/wavefront/agent/core/handlers/AbstractReportableEntityHandler.java

```diff
@@ -0,0 +1,197 @@
+ * @param <U> the type of the output object as handled by {@link SenderTask}
+ */
+abstract class AbstractReportableEntityHandler<T, U> implements ReportableEntityHandler<T> {
+  protected static final String MULTICASTING_TENANT_TAG_KEY = "multicastingTenantName";
+  private static final Logger logger =
+      Logger.getLogger(AbstractReportableEntityHandler.class.getCanonicalName());
+  final QueueInfo queue;
+  final String handler;
+
+  final Function<T, String> serializer;
+  final String rateUnit;
+  final BurstRateTrackingCounter receivedStats;
+  private final Logger blockedItemsLogger;
+  private final Counter receivedCounter;
+  private final Counter blockedCounter;
+  private final Counter rejectedCounter;
+  private final Timer timer;
+
+  @SuppressWarnings("UnstableApiUsage")
+  private final RateLimiter noDataStatsRateLimiter = RateLimiter.create(1.0d / 60);
+
+  /**
+   * @param queue metrics pipeline key (entity type + port number)
+   * @param serializer helper function to convert objects to string. Used when writing blocked
+   *     points to logs.
+   * @param blockedItemsLogger a {@link Logger} instance for blocked items
+   */
+  AbstractReportableEntityHandler(
+      String handler,
+      @NotNull QueueInfo queue,
+      final Function<T, String> serializer,
+      @Nullable final Logger blockedItemsLogger) {
+    this.handler = handler;
+    this.queue = queue;
+    //noinspection UnstableApiUsage
+    this.serializer = serializer;
+    this.rateUnit = queue.getEntityType().getRateUnit();
+    this.blockedItemsLogger = blockedItemsLogger;
+
+    MetricsRegistry registry = Metrics.defaultRegistry();
+    String metricPrefix = queue.getName() + "." + this.handler;
+    MetricName receivedMetricName = new MetricName(metricPrefix, "", "received");
+    this.receivedCounter = registry.newCounter(receivedMetricName);
+    this.blockedCounter = registry.newCounter(new MetricName(metricPrefix, "", "blocked"));
+    this.rejectedCounter = registry.newCounter(new MetricName(metricPrefix, "", "rejected"));
+    this.receivedStats = new BurstRateTrackingCounter(receivedMetricName, registry, 1000);
+
+    registry.newGauge(
+        new MetricName(metricPrefix + ".received", "", "max-burst-rate"),
+        new Gauge<Double>() {
+          @Override
+          public Double value() {
+            return receivedStats.getMaxBurstRateAndClear();
+          }
+        });
+
+    timer = new Timer("stats-output-" + queue.getName() + "." + this.handler);
+    timer.scheduleAtFixedRate(
+        new TimerTask() {
+          @Override
+          public void run() {
+            printStats();
+          }
+        },
+        10_000,
+        10_000);
+    timer.scheduleAtFixedRate(
+        new TimerTask() {
+          @Override
+          public void run() {
+            printTotal();
+          }
+        },
+        60_000,
+        60_000);
+  }
+
+  @Override
+  public void reject(@Nullable T item, @Nullable String message) {
```

Comment:
In the current proxy imp. Reject also increments the blocked counter. Is the change in behavior intended?